docs/source/extending/traits.rst

```diff
@@ -0,0 +1,139 @@
+Traits and TraitsUI
+===================
+
+Traits is a python package, developed by Enthought, for creating and interacting with
+statically-typed variables: '**traits**'. To efficiently develop UIs for traits
+Enthough developed a sister package, TraitsUI.
+
+A class that has traits variables as attributes, inherits from ``HasTraits``,
+so that those variables can be initialized and handled appropriately. Most, if not all,
+of the classes in the BDSS (and in the Workflow Manager) inherit from ``HasTraits``,
+therefore before extending BDSS it is useful to have some basic knowledge of Traits and
+TraitsUI.
+
+Full documentation can be found here:
+
+`Traits <https://docs.enthought.com/traits/>`_
+
+`TraitsUI <https://docs.enthought.com/traitsui/traitsui_user_manual/index.html#contents>`_
+
+These provide brief introductions to the packages. Here we provide an even more
+minimal introduction that should make the code examples in following topics clearer.
+
+Traits
+------
+Traits are class objects (like every variable in python). The more common classes just
+wrap around a built-in python type, with a class name that is the capitalised version
+of the built-in type. For instance, ::
+
+    # initialization of a string trait, x.
+    x = Str('hello world')
+
+    # initialization of a dictionary trait, y.
+    y = Dict({'English':'hello world', 'German':'hallo welt'})
+
+    # initialization of a float trait, z, to the default value of 0.0
+    z = Float()
+    print(z)
+    >> 0.0
```

Comment:
I noticed that sometimes these code blocks do not always get correctly linted by Sphinx. We can use the `.. code-block:: python` block command to explicitly force formatting if needs be (https://www.sphinx-doc.org/en/1.5/markup/code.html).